src/core/thread/data_poll_manager.cpp

```diff
@@ -351,6 +354,23 @@ void DataPollManager::SendFastPolls(uint8_t aNumFastPolls)
     }
 }
 
+otError DataPollManager::StopFastPolls(void)
```

Comment:
Do we expect caller to always match its `StartFastPolls()` call with a corresponding `StopFastPolls()` call
Note that we may not receive a response when we are waiting for it.

Reply:
@abtink, what's the purpose for the caller match?
If the expecting response is not received, Fast Poll mode would automatically stop after tried specific numbers. Here the introduction of `StopFastPolls()` is to end fast poll mode earlier once the expecting response is received, other than always send the specific number of data polls in fast speed.
Thoughts?

Reply:
@rongli, my question/comment was about maintaining the new `mFastPollsUsers` counter (not the fast poll mode itself).

The new code increments it by one from `SendFastPolls()`, and decrements it in `StopFastPolls()`. It's not seemed to be cleared/set from anywhere else (other than set to zero from the constructor).

So if user of `DataPollManager` calls `SendFastPoll()` without a matching later `StopFastPolls()` call, the counter will remain incorrect.  

To avoid the need for mathicng call, I think we need to  clear `mFastPollUsers` when `mRemainingFastPolls` hits zero (i.e., when `DataPollManager` stops the fast poll mode).